src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -2552,8 +2557,88 @@ void Lowering::ContainCheckNeg(GenTreeOp* neg)
 }
 
 //----------------------------------------------------------------------------------------------
-// Try converting SELECT/SELECTCC to CINC/CINCCC. Conversion is possible only if
-// both the trueVal and falseVal are integral constants and abs(trueVal - falseVal) = 1.
+// TryLowerCselToCinvOrCneg: Try converting SELECT/SELECTCC to SELECT_INV/SELECT_INVCC. Conversion is possible only if
+// one of the operands of the select node is inverted.
+//
+// Arguments:
+//     select - The select node that is now SELECT or SELECTCC
+//     cond   - The condition node that SELECT or SELECTCC uses
+//
+void Lowering::TryLowerCselToCinvOrCneg(GenTreeOp* select, GenTree* cond)
+{
+    assert(select->OperIs(GT_SELECT, GT_SELECTCC));
+
+    bool     shouldReverseCondition;
+    GenTree* invertedOrNegatedVal;
+    GenTree* nonInvertedOrNegatedVal;
+    GenTree* nodeToRemove;
+
+    GenTree*   trueVal  = select->gtOp1;
+    GenTree*   falseVal = select->gtOp2;
+    const bool isCneg   = trueVal->OperIs(GT_NEG) || falseVal->OperIs(GT_NEG);
+
+    if (trueVal->OperIs(GT_NOT) || trueVal->OperIs(GT_NEG))
+    {
+        shouldReverseCondition  = true;
+        invertedOrNegatedVal    = trueVal->gtGetOp1();
+        nonInvertedOrNegatedVal = falseVal;
+        nodeToRemove            = trueVal;
+    }
+    else
+    {
+        shouldReverseCondition  = false;
+        invertedOrNegatedVal    = falseVal->gtGetOp1();
+        nonInvertedOrNegatedVal = trueVal;
+        nodeToRemove            = falseVal;
+    }
+
+    if (!cond->OperIsCompare() && select->OperIs(GT_SELECT))
```

Comment:
```suggestion
    if (shouldReverseCondition && !cond->OperIsCompare() && select->OperIs(GT_SELECT))
```
?

Also would be nice to either assert in this function that one of the operands has NOT/NEG, or move the check in here and return a bool on successful transformation (my personal preference, but I will leave it up to you).

Reply:
What I mean was something like:
```cpp
if (!TryLowerCselToCinvOrCneg(select, cond))
  TryLowerCselToCinc(select, cond);
```
inside `LowerSelect`. That way all of the checks are left up to the inner functions. But this looks ok to me as is.